ocaml/libs/tracing/tracing.ml

```diff
@@ -281,6 +281,99 @@ module Spans = struct
 
   let spans = Hashtbl.create 100
 
+  module SpanFrequencies = struct
+    let depth = 10
+
+    let frequency_tbl = Hashtbl.create 100
+
+    let incr_frequency_tbl ~(span : Span.t) =
+      match Hashtbl.find_opt frequency_tbl span.name with
+      | None ->
+          Hashtbl.add frequency_tbl span.name 1
+      | Some v ->
+          Hashtbl.replace frequency_tbl span.name (succ v)
+
+    let decr_frequency_tbl ~(span : Span.t) =
+      match Hashtbl.find_opt frequency_tbl span.name with
+      | None ->
+          ()
+      | Some v ->
+          if pred v = 0 then
+            Hashtbl.remove frequency_tbl span.name
+          else
+            Hashtbl.replace frequency_tbl span.name (pred v)
+
+    module SpanName = struct
+      type t = string
+
+      let compare = String.compare
+    end
+
+    module SpanMap = Map.Make (SpanName)
+
+    type frequencies = (string * int) list [@@deriving yojson]
+
+    let take_most n k v acc =
+      let last_min = SpanMap.min_binding_opt acc in
+      match last_min with
+      | None ->
+          SpanMap.add k v acc
+      | Some last_min ->
+          if SpanMap.cardinal acc < n then
+            SpanMap.add k v acc
+          else if v <= snd last_min then
+            acc
+          else
+            SpanMap.remove (fst last_min) acc |> SpanMap.add k v
```

Comment:
This could use `Map.update`.

Reply:
At first I thought I could use it as well but I am replacing the binding with a new one with a different key.